parlai/tasks/hotpotqa/agents.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import ParlAIDialogTeacher
+from parlai.core.utils import warn_once
+from .build import build
+
+import copy
+import os
+
+
+def _path(opt):
+    # Build the data if it doesn't exist.
+    build(opt)
+    dt = opt['datatype'].split(':')[0]
+    if dt == 'test_distractor':
```

Comment:
Note that these are invalid datatypes for ParlAI (see [here](https://github.com/facebookresearch/ParlAI/blob/master/parlai/core/params.py#L393) for valid ones). One way to resolve this would be to have a separate teacher for each set (like in the [ConvAI2 teachers](https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/convai2/agents.py), which pass an arg to `_path` to get the appropriate dataset). Or, you can override the static method `add_cmdline_args` to add another argument that specifies which set to use (like [here](https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/personality_captions/agents.py#L138))